src/Sulu/Bundle/WebsiteBundle/Controller/DefaultController.php

```diff
@@ -64,4 +61,64 @@ public function redirectAction(Request $request)
     {
         return new RedirectResponse($request->get('url'), 301);
     }
+
+    /**
+     * Resolve the redirect URL, appending any additional path data
+     *
+     * @param string $url Original webspace URI
+     * @param string $redirectUrl  Redirect webspace URI
+     * @param string $requestUri The actual incoming request URI
+     *
+     * @return string URL to redirect to
+     */
+    protected function resolveRedirectUrl($redirectUrl, $requestUri)
+    {
+        $redirectInfo = $this->parseUrl($redirectUrl);
+        $requestInfo = $this->parseUrl($requestUri);
+
+        $url = sprintf('%s://%s', $requestInfo['scheme'], $redirectInfo['host']);
+
+        if (isset($requestInfo['port'])) {
+            $url .= ':' . $requestInfo['port'];
+        }
+
+        if (isset($redirectInfo['path'])) {
+            $url .= $redirectInfo['path'];
+        }
+
+        if (isset($requestInfo['path'])) {
```

Comment:
given `/sulu.lo/foobar` with a portal `sulu.io/en` then the request info path is `/foobar` and should be added to `/en`